### PR TITLE
Consolidate function definitions

### PR DESCRIPTION
**Pull request by volkangurel**
Consolidate function definitions

This PR is one of several to split https://github.com/layerai/sdk/pull/127 up. This PR should merge after #137 is merged.

### CODE REVIEW

File: layer/decorators/layer_wrapper.py [hunk 1]
```diff
@@ -62,3 +65,44 @@ def __init__(
...
+                if d.project_name is not None:
+                    full_path_dep = d.with_project_full_name(
+                        ProjectFullName(
+                            account_name=current_project_full_name.account_name,
+                            project_name=d.project_name,
+                        )
+                    )
+                else:
+                    full_path_dep = d.with_project_full_name(current_project_full_name)
+            asset_dependencies.append(full_path_dep)
```
Comment (cihati) [inline layer/decorators/layer_wrapper.py:87]: nit: I'd put this into a separate method. same for get_pip_dependencies below

File: test/unit/projects/test_project_hash_calculator.py [hunk 2]
```diff
@@ -55,3 +59,20 @@ def func3():
 
     # then
     assert first_hash != second_hash
+
+
+def _make_test_function_definition(
```
Comment (cihati) [inline test/unit/projects/test_project_hash_calculator.py:64]: https://github.com/layerai/sdk/pull/127#discussion_r901118767

File: layer/contracts/runs.py [hunk 3]
```diff
@@ -198,53 +176,8 @@ def _pack(self) -> None:
...
             cloudpickle.dump(self.func, file, protocol=pickle.DEFAULT_PROTOCOL)
 
-        # Add requirements to tarball
-        if self.pip_requirements_file:
-            shutil.copy(self.pip_requirements_file, self.environment_path)
-        elif self.pip_packages:
-            with open(self.environment_path, "w") as reqs_file:
-                reqs_file.writelines(
-                    list(map(lambda package: f"{package}\n", self.pip_packages))
-                )
```
Comment (cihati) [inline layer/contracts/runs.py:208]: you seem to be changing this part as part of this refactor -- did you test this thoroughly?

File: layer/contracts/runs.py [hunk 4]
```diff
@@ -61,41 +60,41 @@ def local_relative_paths(self) -> Iterator[str]:
...
-        self.resource_paths = layer_settings.get_resource_paths()
-        fabric = layer_settings.get_fabric()
-        if fabric is None:
-            fabric = Fabric.default()
-        self._fabric = fabric
-        self._dependencies: List[AssetPath] = layer_settings.get_dependencies()
-        self.pip_packages = layer_settings.get_pip_packages()
-        self.pip_requirements_file = layer_settings.get_pip_requirements_file()
 
         self.func_source = inspect.getsource(self.func)
```
Comment (cihati) [inline layer/contracts/runs.py:100]: so, the `source_code_digest`, also `uri` above etc are all dataset-specific. I guess one downside of this "bundling" is that models will have these unnecessarily as well..

**Review by cihati — APPROVED** [review]

**Comment by cihati** [comment]
@volkangurel I think this got automerged, but please take a note of my comments = )

File: layer/decorators/layer_wrapper.py (hunk 1 above)
Comment (volkangurel) [reply layer/decorators/layer_wrapper.py:87]: done

File: test/unit/projects/test_project_hash_calculator.py (hunk 2 above)
Comment (volkangurel) [reply test/unit/projects/test_project_hash_calculator.py:64]: done

File: layer/contracts/runs.py (hunk 4 above)
Comment (volkangurel) [reply layer/contracts/runs.py:100]: yes, true, but with several more factors we should be able to get datasets and models closer and more consistent with each other. for example, no reason why `source_code_digest` and `uri` can't be attributes of both datasets and models.

File: layer/contracts/runs.py (hunk 3 above)
Comment (volkangurel) [reply layer/contracts/runs.py:208]: not sure what is thoroughly, but tested with several remote execution examples that I think should cover this.

**Pull request merged**